feat(examples): Refactor examples: move files to GraphList, GraphMatrix and Experimental.

Ghoulayush opened the pull request:
…; remove extras folder

## Which issue does this PR close?
- Closes #102 .

## Rationale for this change
This change reorganizes the example files into a more focused and structured layout by moving them into two top-level folders `GraphList/` and `GraphMatrix/` and an `experimental/` folder for less central examples.  
The `extras/` folder was removed after migrating its contents to the `experimental/` folder to avoid scattering examples and to improve maintainability.

## What changes are included in this PR?
- Moved example files into `examples/GraphList/`, `examples/GraphMatrix/`, and `examples/experimental/`.
- Removed the `extras/` folder.
- Prepared the structure to support easier creation of dedicated `_usage.cpp` files per function in future work.

## Are these changes tested?
No functional code changes made, only reorganized files. Existing tests remain unchanged and cover regressions.

## Are there any user-facing changes?
No user-facing changes or API modifications introduced in this PR.

Ghoulayush committed:
Refactor examples: move files to GraphList, GraphMatrix, experimental; remove extras folder

Ghoulayush commented:
@SharonIV0x86 @Sayantani9  I have opened this PR addressing #102 , moving the examples as discussed. The refactor is not complete yet, and I will continue updating the usage files.

Sayantani9 commented:
@Ghoulayush Kindly Update your PR's status!

Ghoulayush commented:
Hi @Sayantani9 , Sorry for not updating earlier. My exams are now over, so I’ll complete the pending work and update the PR by tomorrow.

Sayantani9 commented:
Hello @Ghoulayush try to Update ASAP. Then we'll be able to proceed with the merge.

Ghoulayush committed:
Include demo usage examples for GraphList operations

Ghoulayush commented:
@Sayantani9 , I have added demo examples for GraphList functions. I will soon add usage examples for GraphMatrix functions and will update the necessary changes in CMakeLists.txt.

Ghoulayush committed:
Merge upstream/main into refactor-examples-layout

SharonIV0x86 commented:
@Ghoulayush Try to update the cmake file to adjust it with the updated examples structure.

SharonIV0x86 commented on .vscode/settings.json:
```diff
@@ -0,0 +1,81 @@
+{
```
Kindly remove the ``.vscode`` folder.

Sayantani9 replied:
@Ghoulayush kindly update the work as per the guidance of @SharonIV0x86 with the discussed necessary changes & update us so we can proceed!

Ghoulayush replied:
@SharonIV0x86 i have removed the `.vscode` folder

Ghoulayush committed:
Include demo usage examples for GraphMatrix Operations and did necesarry changes in CmakeLists.txt

Ghoulayush committed:
Removed the .vscode folder

Ghoulayush committed:
Merge upstream/main into refactor-examples-layout

Ghoulayush commented:
@Sayantani9 @SharonIV0x86  I have added the demo usage examples for GraphMatrix Functions as well as have done the necessary changes in the CMakeLists.txt and removed the `.vscode` folder

SharonIV0x86 committed:
fix: fixed failing build by excluding problematic example files.

SharonIV0x86 commented:
@Ghoulayush I have fixed the failing build. The problem was due to these 3 example files.
- ``examples/GraphMatrix/operator_usage.cpp``
- ``examples/GraphMatrix/updateEdge_usage.cpp``
- ``examples/GraphMatrix/hasVertex_usage.cpp``
I have commented out code in these example files because the API function calls are not being made correctly so you need to fix these.

There are a few changes i have done.
- Fixed CMake file.
- Renamed ``examples/GraphMatrix/operator[]_usage.cpp`` to ``operator_usage.cpp``
- Created ``/example/Policies`` folder to hold the example files related to Policies.

The building and testing shall pass now and we can fix the formatting/linting later its no big issue.

Ghoulayush commented:
Thanks @SharonIV0x86 for fixing the build and pointing out the issues! I’ll go through the three example files and correct the API function calls. Also noted the CMake changes, renaming, and the new Policies folder. I’ll make the necessary fixes and update accordingly.

SharonIV0x86 committed:
fix: fixed examples and formatting

SharonIV0x86 reviewed (approved):
Approved.

SharonIV0x86 commented:
@Ghoulayush Thank you for contributing your PR has been merged, kindly ⭐ the repository as it shows appreciation to the project maintainers.

Ghoulayush commented:
Thank you for reviewing and merging my PR! @SharonIV0x86, I’ve starred the repo as well—happy to contribute.
